src/App.tsx

```diff
@@ -15,36 +15,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+const SOTR_BY_ALPHABET: string = 'alphabet';
+const SOTR_BY_LENGTH: string = 'length';
+
+function SortGoodsFrom(
+  originArray: string[],
+  reverse: boolean,
+  sortBy: string,
+) {
+  let mutateArray: string[] = [...originArray];
+
+  switch (sortBy) {
+    case SOTR_BY_ALPHABET:
+      mutateArray = mutateArray.sort((a, b) => a.localeCompare(b));
+      break;
+    case SOTR_BY_LENGTH:
+      mutateArray = mutateArray.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      break;
+  }
+
+  if (reverse) {
+    mutateArray.reverse();
+  }
+
+  return mutateArray;
+}
+
 export const App: React.FC = () => {
+  const [isReverse, setIsReverse] = useState(false);
+  const [sort, setSort] = useState('');
+
+  const visibleGoods: string[] = SortGoodsFrom(
```

Comment:
```suggestion
  const visibleGoods = SortGoodsFrom(
```
You don't have to specify type every time, typescript understands what you return from the function